### PR TITLE
Jpd defra/issue186

### LINKED ISSUE #186: Add Local Development Guide section to README
As is shown in `pyspark_vector_files`, a guide for local development of the data model package should be added to the README. This will contain information on setting up an appropriate development environment, e.g. with poetry and GDAL. Also note that for use of Spark in a development environment, Java is required, and should be installed as shown in [this link](https://www.digitalocean.com/community/tutorials/how-to-install-java-with-apt-on-ubuntu-22-04).

### PR DESCRIPTION
**Pull request by jpd-defra**
Simple Development Guide, mostly lifted from `pyspark_vector_files` with added Java section and reformatted from md to rst. 

Fixes #186

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -28,6 +28,101 @@ Usage
...
+
+Verify the installation:
+
+.. code:: console
+
+   ogrinfo --version
+   # GDAL 3.4.3, released 2022/04/22
+
+
+Install `poetry 1.1.13`
```
Comment (EFT-Defra) [inline README.rst]: This should now be `poerty 1.2`

```suggestion
Install `poetry 1.2`
```

File: README.rst [hunk 2]
```diff
@@ -28,6 +28,101 @@ Usage
...
+++++++++++++
+
+Java is required for Spark to work correctly. This guide details `Java installation on Ubuntu <https://www.digitalocean.com/community/tutorials/how-to-install-java-with-apt-on-ubuntu-22-04>`__.
+
+The required commands are:
+
+.. code:: console
+
+   sudo apt install default-jre
+   sudo apt install default-jdk
```
Comment (EFT-Defra) [inline README.rst]: This can be shortened to:

```suggestion
   sudo apt install default-jre \
     default-jdk
```

**Review by EFT-Defra — CHANGES_REQUESTED** [review]
Two minor changes, please

**Comment by jpd-defra** [comment]
Thanks for reviewing. Changes implemented.

**Pull request merged**